gatsby-config.js

```diff
@@ -8,198 +10,211 @@ require("dotenv").config({
+          options: {
+            maxWidth: 1035,
+            // sizeByPixelDensity: true,
+            markdownCaptions: true,
+            showCaptions: false,
+            linkImagesToOriginal: false,
           },
-          { // Adds titles to code blocks... obvi
-            resolve: `gatsby-remark-code-titles`, // https://www.gatsbyjs.com/plugins/gatsby-remark-code-titles/
-            options: {
-              className: `gatsby-remark-code-title`, // Defines the CSS class it uses, for custom styling
-            },
+        },
+        { // Adds titles to code blocks... obvi
+          resolve: `gatsby-remark-code-titles`, // https://www.gatsbyjs.com/plugins/gatsby-remark-code-titles/
+          options: {
+            className: `gatsby-remark-code-title`, // Defines the CSS class it uses, for custom styling
           },
-          `gatsby-remark-static-images`,
-          `gatsby-remark-autolink-headers`,
-          {
-            resolve: `gatsby-remark-prismjs`,
-            options: {
-              classPrefix: "language-",
-              inlineCodeMarker: null,
-              noInlineHighlight: true,
-              aliases: {},
-              prompt: {
-                user: "user",
-                host: "localhost",
-              },
+        },
+        `gatsby-remark-static-images`,
+        `gatsby-remark-autolink-headers`,
+        {
+          resolve: `gatsby-remark-prismjs`,
+          options: {
+            classPrefix: "language-",
+            inlineCodeMarker: null,
+            noInlineHighlight: true,
+            aliases: {},
+            prompt: {
+              user: "user",
+              host: "localhost",
             },
           },
-          {
-            resolve: "gatsby-remark-external-links",
-            options: {
-              target: null,
-              rel: "nofollow noopener external",
-            },
+        },
+        {
+          resolve: "gatsby-remark-external-links",
+          options: {
+            target: null,
+            rel: "nofollow noopener external",
           },
-          `gatsby-remark-heading-slug`,
-        ],
-      },
-    },
-    `gatsby-transformer-sharp`,
-    `gatsby-plugin-sharp`,
-    {
-      resolve: "gatsby-plugin-web-font-loader",
-      options: {
-        google: {
-          families: ["Open Sans"],
         },
-      },
+        `gatsby-remark-heading-slug`,
+      ],
     },
-    `gatsby-plugin-react-helmet`,
-    {
-      resolve: 'gatsby-plugin-sitemap',
+  },
+  `gatsby-transformer-sharp`,
+  `gatsby-plugin-sharp`,
+  {
+    resolve: "gatsby-plugin-web-font-loader",
+    options: {
+      google: {
+        families: ["Open Sans"],
+      },
     },
-    `gatsby-plugin-fontawesome-css`,
-  ],
+  },
+  `gatsby-plugin-react-helmet`,
+  {
+    resolve: 'gatsby-plugin-sitemap',
+  },
+  `gatsby-plugin-fontawesome-css`,
+]
+
+if(config.search && config.search.enabled && config.search.algoliaAppId && config.search.algoliaAdminKey) {
```

Comment:
let's add this validation to a const with a meaningful name and add that to the if statement
```suggestion
const some_meaningful_name =  config.search && config.search.enabled && config.search.algoliaAppId && config.search.algoliaAdminKey
if(some_meaningful_name) {
```